codec/src/main/java/io/netty/handler/codec/DefaultTextHeaders.java

```diff
@@ -396,4 +404,217 @@ public TextHeaders clear() {
+
+        TextHeaders set(CharSequence name, CharSequence... values);
+        TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values);
+
+        TextHeaders setObject(CharSequence name, Object... values);
+        TextHeaders setObject(CharSequence name, Iterable<?> values);
+    }
+
+    /*
+     * Will add multiple values for the same header as multiple separate headers.
+     */
+    private final class MultipleFieldsValueComposer implements ValuesComposer {
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence value) {
+            DefaultTextHeaders.super.add(name, value);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.add(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders add(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.add(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.addObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders addObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.addObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, CharSequence... values) {
+            DefaultTextHeaders.super.set(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders set(CharSequence name, Iterable<? extends CharSequence> values) {
+            DefaultTextHeaders.super.set(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Object... values) {
+            DefaultTextHeaders.super.setObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+
+        @Override
+        public TextHeaders setObject(CharSequence name, Iterable<?> values) {
+            DefaultTextHeaders.super.setObject(name, values);
+            return DefaultTextHeaders.this;
+        }
+    }
+
+    /**
+     * Will add multiple values for the same header as single header with a comma separated list of values.
+     *
+     * Please refer to section <a href="https://tools.ietf.org/html/rfc7230#section-3.2.2">3.2.2 Field Order</a>
+     * of RFC-7230 for details.
+     */
+    private final class SingleHeaderValuesComposer implements ValuesComposer {
+
+        private final ValueConverter<CharSequence> valueConverter = valueConverter();
+        private final CsvValueQuoter<Object> objectQuoter = new CsvValueQuoter<Object>() {
```

Comment:
Lets use `escape` instead of `quote`.  I think this is a global comment for this PR.